env0/resource_environment.go

```diff
@@ -255,18 +263,23 @@ func resourceEnvironmentCreate(ctx context.Context, d *schema.ResourceData, meta
 			return diagErr
 		}
 
+		createTemplatePayload.IsSingleUse = true
+
 		name := d.Get("name").(string)
 		createTemplatePayload.Name = "single-use-template-for-" + name
 
-		template, err := apiClient.TemplateCreate(createTemplatePayload)
+		template, err = apiClient.TemplateCreate(createTemplatePayload)
```

Comment:
So we took it internally, and really didn't like the fact that the client needs to create the template and not the backend. We're going to do a quick refactor where we'll create a new API for "environment create with template" that would get all relevant parameters.

This means that this issue is **blocked** until we change it on our end. I believe it's going to take a couple of days and will probably require a lil' bit refactor from your end. I'm sorry for the trouble 🙏

Reply:
sounds good.
Thanks for the update!